pkg/solana/txm/txm.go

```diff
@@ -417,24 +417,26 @@ func (txm *Txm) confirm() {
 						)
 
 						// check confirm timeout exceeded
-						if txm.txs.Expired(s[i], txm.cfg.TxConfirmTimeout()) {
-							id, err := txm.txs.OnError(s[i], txm.cfg.TxRetentionTimeout(), TxFailDrop)
+						if txm.cfg.TxConfirmTimeout() != 0*time.Second && txm.txs.Expired(s[i], txm.cfg.TxConfirmTimeout()) {
+							id, err := txm.txs.OnError(s[i], txm.cfg.TxRetentionTimeout(), Errored, TxFailDrop)
 							if err != nil {
-								txm.lggr.Infow("failed to mark transaction as errored", "id", id, "signature", s[i], "timeoutSeconds", txm.cfg.TxConfirmTimeout(), "error", err)
+								txm.lggr.Errorw("failed to mark transaction as errored", "id", id, "signature", s[i], "timeoutSeconds", txm.cfg.TxConfirmTimeout(), "error", err)
 							} else {
-								txm.lggr.Infow("failed to find transaction within confirm timeout", "id", id, "signature", s[i], "timeoutSeconds", txm.cfg.TxConfirmTimeout())
+								txm.lggr.Debugw("failed to find transaction within confirm timeout", "id", id, "signature", s[i], "timeoutSeconds", txm.cfg.TxConfirmTimeout())
 							}
 						}
 						continue
 					}
 
 					// if signature has an error, end polling
 					if res[i].Err != nil {
-						id, err := txm.txs.OnError(s[i], txm.cfg.TxRetentionTimeout(), TxFailRevert)
-						if err != nil {
-							txm.lggr.Infow("failed to mark transaction as errored", "id", id, "signature", s[i], "error", err)
-						} else {
-							txm.lggr.Debugw("tx state: failed", "id", id, "signature", s[i], "error", res[i].Err, "status", res[i].ConfirmationStatus)
+						if txState, errType := txm.processError(s[i], res[i].Err, false); errType != NoFailure {
```

Comment:
in which case we will have an err of NoFailure type? what does it means?

Reply:
nvm, I found the explanation in processError func

Reply:
maybe useful to add a comment here anyway